mediation-old/src/main/java/com/wso2telco/dep/mediator/internal/ResourceURLUtil.java

```diff
@@ -209,12 +209,16 @@ private String findAPIType(String ResourceURL) {
 	public String[] getParamPairs(String url) throws UnsupportedEncodingException {
 
 		List<String> paramPairs = new ArrayList<String>();
-		String[] urlParts = url.split("\\?");
+		String[] urlParts = url.split("\\?",2);
```

Comment:
As per the specification, MSISDN must be url escaped value
"The MSISDN, of the mobile device to locate. Repeat the
address parameter for multiple devices. 
The protocol and ‘+’ identifier must be used for MSISDN, and must be URL-escaped. 
%3A represents ‘:’ and %2B represents ‘+"

Hence I think returned error response is a expected behavior. Hence its good to recheck bug with the specification.

Reply:
WIll update the spec to comply with this.

Reply:
As mentioned by Charith this should be backward compatible